org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/JDTLanguageServer.java

```diff
@@ -319,9 +330,83 @@ public void didChangeConfiguration(DidChangeConfigurationParams params) {
 			JavaLanguageServerPlugin.logException(e.getMessage(), e);
 		}
 		pm.registerWatchers();
+		configureFormatter();
 		logInfo(">>New configuration: " + settings);
 	}
 
+	public void configureFormatter() {
+		String formaterUrl = preferenceManager.getPreferences().getFormatterUrl();
+		Map<String, String> options = null;
+		if (formaterUrl != null) {
+			URL url = null;
+			try {
+				url = new URL(formaterUrl);
+			} catch (MalformedURLException e1) {
+				File file = new File(formaterUrl);
+				if (file.isFile()) {
+					try {
+						url = file.toURI().toURL();
+					} catch (MalformedURLException e) {
+						JavaLanguageServerPlugin.logInfo("Invalid formatter:" + formaterUrl);
+					}
+				} else {
+					unregisterWatcher(file);
```

Comment:
you'll miss this branch if
- formaterUrl is a file:// url
- formaterUrl is now empty but was not before